tests/gold_tests/headers/http103.test.py

```diff
@@ -0,0 +1,54 @@
+'''
+Test the 103 Early Hints Response
+'''
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+import subprocess
+
+Test.Summary = '''
+Test the 103 Early Hints Response
+'''
+
+# need Curl
+Test.SkipUnless(
+    Condition.HasProgram("curl", "Curl need to be installed on system for this test to work")
+)
+
+Test.ContinueOnFail = True
+HTTP_103_HOST = 'www.103earlyhints.test'
+
+# Origin Server
+server = Test.MakeOriginServer("server")
+request_header = {"headers": "GET / HTTP/1.1\r\nHost: {0}\r\n\r\n".format(HTTP_103_HOST), "timestamp": "1469733493.993", "body": ""}
+response_header = {"headers": "HTTP/1.1 103 Early Hints\r\nLink: </style.css>; rel=preload; as=style\r\n\r\nHTTP/1.1 200 OK\r\nConnection: close\r\n\r\n", "timestamp": "1469733493.993", "body": ""}
```

Comment:
@persiaAziz Does uServer support multiple HTTP messages?

Reply:
From wireshark, uServer returns below. The status line looks wired.
```
HTTP/1.1 103 \r\n
Link: </style.css>; rel=preload; as=style\r\n
\r\n
HTTP/1.1 400 Bad Request\r\n
Connection: close\r\n
\r\n
```

Reply:
Run uWServer directly and got below. It looks like we also need 103 support on uWServer :)
```
started server
Error (<class 'IndexError'>, IndexError('list index out of range',), <traceback object at 0x10bc195c8>) Host: 127.0.0.1:52858
User-Agent: curl/7.54.0
Accept: */*
```

Reply:
Are uWServer using `HTTPStatus` of python when sending HTTP response? We need 103 support here. 
https://github.com/python/cpython/blob/3.6/Lib/http/__init__.py#L27